### PR TITLE
Bumped OpenTelemetry Collector to v0.13.0

### PR DESCRIPTION
**Pull request by dengliming**
closes [https://github.com/open-telemetry/opentelemetry-operator/issues/93](https://github.com/open-telemetry/opentelemetry-operator/issues/93)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,11 @@
 Changes by Version
 ==================
 
+0.13.0 (2020-10-21)
```
Comment (jpkrohling) [inline CHANGELOG.md:4]: Could you change it to 22? I'll double-check the release notes tomorrow and merge/release.

**Review by jpkrohling — COMMENTED** [review]
Thanks for taking care of this one!

File: CHANGELOG.md (hunk 1 above)
Comment (dengliming) [reply CHANGELOG.md]: sure. It was done. Thanks.

**Review by jpkrohling — COMMENTED** [review]
LGTM. I'll check #100 before releasing this, in case it's a real bug that needs fixing.

**Comment by jpkrohling** [comment]
@mergify rebase

**Comment by jpkrohling** [comment]
Checking the release notes, we need to handle the change open-telemetry/opentelemetry-collector#1891. Either we need to replace the image with the contrib one (v0.13.1, which includes this processor), or remove the processor. In any case, we need to add a message to the status object.

@objectiser, what do you think would be more appropriate?

**Comment by objectiser** [comment]
@jpkrohling when you say "remove the processor", remove it from where?

**Comment by jpkrohling** [comment]
From the CR's configuration. 

This processor (`groupbytrace`) has been removed from the core distribution and added to the contrib on v0.13.1. CRs that are configured using this processor will fail if we don't change the configuration during the upgrade, so, we should either automatically remove it from the CR, or change the CR's image to use contrib.

**Comment by objectiser** [comment]
@jpkrohling As this is pre-1.0, and components are still be shuffled between core and contrib, not sure the operator should take care of all these situations yet?

**Comment by jpkrohling** [comment]
The code itself shouldn't be that hard once we decide what's the best approach, and this might help guide future decisions. I'd be OK in just adding something to the changelog as well :-)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,11 @@
 Changes by Version
 ==================
 
+0.13.0 (2020-10-22)
+-------------------
+
+* Bumped OpenTelemetry Collector to v0.13.0 ([#101](https://github.com/open-telemetry/opentelemetry-operator/pull/101), [@dengliming](https://github.com/dengliming))
```
Comment (jpkrohling) [inline CHANGELOG.md:7]: Looks like there's another change that is relevant:

```Markdown
* Allow for spec.Env to be set on the OTEL Collector Spec ([#94](https://github.com/open-telemetry/opentelemetry-operator/pull/94), [@ekarlso](https://github.com/ekarlso))
```

**Comment by objectiser** [comment]
@jpkrohling yeah we need to have a policy - although one of the issues is that users can specify their own custom image (whether contrib or something else) and the operator will only be able to deal with what it knows about - just depends whether that is just core, or should it also cover contrib components?

**Comment by jpkrohling** [comment]
What I had in mind is to add a "compatibility" flag to the new CR (https://github.com/open-telemetry/opentelemetry-operator/issues/77#issuecomment-702339135), stating how we should treat the CR for upgrades.

**Comment by jpkrohling** [comment]
Given that we don't currently know what's the distribution people are using, it's probably safer to ignore this upgrade for this release.

@dengliming, would you please then add a mention in the changelog that people have to manually deal with this move?

**Comment by dengliming** [comment]
_The groupbytrace processor already be moved to the contrib repository. Either we need to replace the image with the contrib one (v0.13.1, which includes this processor), or remove the processor._

what about this?

**Comment by jpkrohling** [comment]
Sounds good, just a small change:

```Markdown
The `groupbytrace` processor was moved to the contrib repository, requiring a manual intervention in case this processor is being used: either replace the image with the contrib one (v0.13.1, which includes this processor), or remove the processor.
```

**Review by jpkrohling — APPROVED** [review]
LGTM. Thanks again for your help, @dengliming! The new release is coming right away.

**Pull request merged**